app/src/main/java/org/zalando/nakadi/service/job/DlqRedriveEventTypeAttachmentJob.java

```diff
@@ -0,0 +1,161 @@
+        } catch (final Exception e) {
+            LOG.error("Failed to run the job to attach DLQ redrive event type to corresponding subscriptions", e);
+        }
+    }
+
+    private void attachDlqRedriveEventTypeLocked() {
+
+        final List<String> orderedPartitions = eventTypeCache.getOrderedPartitions(dlqRedriveEventTypeName);
+
+        final List<NakadiCursor> beginCursors = orderedPartitions.stream()
+                .map(p -> new Cursor(p, Cursor.BEFORE_OLDEST_OFFSET))
+                .map(c -> cursorConverter.convert(dlqRedriveEventTypeName, c))
+                .collect(Collectors.toList());
+
+        // it's inefficient to re-consume from begin, but we expect this event type to be empty most of the time anyway
+        final HighLevelConsumer consumer = timelineService.createEventConsumer(JOB_NAME + "-job", beginCursors);
+
+        final Set<String> subscriptionIds = new HashSet<>();
+        while (true) {
+            final List<ConsumedEvent> events = consumer.readEvents();
+            if (events.isEmpty()) {
+                // consumed till the current end
+                break;
+            }
+            events.forEach(e -> {
+                        final Map<HeaderTag, String> tags = e.getConsumerTags();
+                        if (tags != null) {
+                            final String sid = tags.get(HeaderTag.CONSUMER_SUBSCRIPTION_ID);
+                            if (sid != null) {
+                                subscriptionIds.add(sid);
+                            }
+                        }
+                    });
+        }
+
+        final List<Partition> unassignedDlqPartitions = orderedPartitions.stream()
+                .map(p -> new Partition(dlqRedriveEventTypeName, p, null, null, Partition.State.UNASSIGNED))
+                .collect(Collectors.toList());
+
+        // for every subscription update the topology, if needed, so that the dlq redrive event type is included
+        subscriptionIds.forEach(sid -> {
+                    try {
+                        addDlqPartitionsToSubscription(
+                                subscriptionRepository.getSubscription(sid),
+                                unassignedDlqPartitions);
+                    } catch (final Exception e) {
+                        // subscription is already gone, but events for redrive are still there; or failed to update
+                        // topology, etc.
+                        LOG.warn("Failed to add DLQ redrive event type to subscription: {}", sid, e);
+                    }
+                });
+    }
+
+    private void addDlqPartitionsToSubscription(
+            final Subscription subscription,
+            final List<Partition> unassignedDlqPartitions) throws Exception {
+
+        final ZkSubscriptionClient client = subscriptionClientFactory.createClient(subscription);
+
+        // idempotent, does not overwrite existing offsets
+        client.createOffsetZNodes(
```

Comment:
maybe we would like to aggregate it under one method in zookeeper client, it looks too low level for a job